### PR TITLE
LogPipe: Improved debug logs 

### PR DESCRIPTION
**Pull request by mitzkia**
Adding improved debug log for parsers, rewrite rules and logsources.

**Motivation came from**: 
 * after adding new config elements like: default-network-drivers, ewmm, app-parser..., syslog-ng's debug log became hard readable
 * improved debug logs of filters 9e3415b4e49b1addc8fac6eccd9f9c814e395e3d

**Conception**:
 1. Log debug message at the beginning and at the end of queuing methods, like: 
    * log_source_queue(), 
    * log_parser_queue(), 
    * log_rewrite_queue()
 * Debug message should contains:
    * name of the relevant logpipe element
    * rule name
    * location
    * msg pointer
    * result of filter_expr_eval_root() or log_parser_process_message() -> only at the end of queuing methods
 * Example:
```
>>>>>> parser rule evaluation begin; rule='#anon-parser0', location='parser generator app-parser:5:15', msg='0x7fc2f4013e00'
...
<<<<<< parser rule evaluation result; result='UNMATCHED', rule='#anon-parser0', location='parser generator app-parser:5:15', msg='0x7fc2f4013e00'
```

 2. Log a debug message from parsers at the beginning and on failures
 * Example
```
json-parser started message processing; input='AAAAAA', prefix='.cim.', marker='@cim:', msg='0x7fc2f4013e00'
json-parser failed; error='marker not found', input='AAAAAA', marker='@cim:'
```

 3. Log a debug message when a message macro will get a value during:
     * log_msg_parse_legacy()
     * log_msg_parse_syslog_proto()
     * syslog_format_handler()

 * Example:
```
Initial message parsing;
Setting macros on BSD syslog message;
Setting value; macro='.unix.pid', value='3763', msg='0x7ff6c0013e00'
Setting value; macro='.unix.uid', value='1000', msg='0x7ff6c0013e00'
Setting value; macro='.unix.gid', value='1000', msg='0x7ff6c0013e00'
```

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/parser/parser-expr.h [hunk 1]
```diff
@@ -52,6 +52,8 @@ log_parser_deinit_method(LogPipe *s)
   return TRUE;
 }
 
+EVTTAG *parser_result_tag(gboolean res);
```
Comment (furiel) [inline lib/parser/parser-expr.h]: Does this need to be part of the public interface? I think this function can be static. Similar comment to the other two variants.

File: modules/dbparser/dbparser.c [hunk 2]
```diff
@@ -218,10 +220,9 @@ log_db_parser_process(LogParser *s, LogMessage **pmsg, const LogPathOptions *pat
 
   if (self->drop_unmatched && !matched)
     {
-      msg_debug("db-parser() failed to parse its input and drop-unmatched flag was specified",
-                evt_tag_str("input", input),
-                log_pipe_location_tag(&s->super),
```
Comment (furiel) [inline modules/dbparser/dbparser.c]: Is this tag accidentally removed?

File: modules/python/python-logparser.c [hunk 3]
```diff
@@ -182,10 +182,10 @@ python_parser_process(LogParser *s, LogMessage **pmsg, const LogPathOptions *pat
   {
     LogMessage *msg = log_msg_make_writable(pmsg, path_options);
 
-    msg_debug("Invoking the Python parse() method",
+    msg_debug("python-parser message processing started",
+              evt_tag_str ("input", input),
               evt_tag_str("parser", self->super.name),
               evt_tag_str("class", self->class),
-              log_pipe_location_tag(&self->super.super),
```
Comment (furiel) [inline modules/python/python-logparser.c]: Why is the location tag deleted here?

File: modules/dbparser/dbparser.c (hunk 2 above)
Comment (mitzkia) [reply modules/dbparser/dbparser.c]: No it was intentional, there would be a redundancy. From now log_pipe_location_tag() would be printed in every case from the very first place: log_source_queue(), log_parser_queue(), log_rewrite_queue(), log_filter_queue().

File: modules/python/python-logparser.c (hunk 3 above)
Comment (mitzkia) [reply modules/python/python-logparser.c]: In my previous comment I have described why it is deleted from here.

File: lib/parser/parser-expr.h (hunk 1 above)
Comment (mitzkia) [reply lib/parser/parser-expr.h]: Agreed, I will change this.

Comment (mitzkia) [reply lib/parser/parser-expr.h]: Fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/logmsg/logmsg.c [hunk 4]
```diff
@@ -1212,6 +1212,10 @@ log_msg_clone_cow(LogMessage *msg, const LogPathOptions *path_options)
   memcpy(self, msg, sizeof(*msg));
   msg->allocated_bytes = allocated_bytes;
 
+  msg_debug("Message was cloned",
```
Comment (furiel) [inline lib/logmsg/logmsg.c]: This should be trace imo.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
Sometimes the fact that the message was cloned can be important to
understand debug logs. We include the address of the message for the very
same reason, but that's not that appearent.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**